x/gov/keeper/deposit.go

```diff
@@ -162,6 +162,44 @@ func (keeper Keeper) AddDeposit(ctx sdk.Context, proposalID uint64, depositorAdd
 	return activatedVotingPeriod, nil
 }
 
+// BurnAndSendDepositsToCommunityPool will burn the (deposits * prposal_cancel_burn_rate) amount of proposal deposits
+// and send remaining deposits of the proposal to community pool.
+func (keeper Keeper) BurnAndSendDepositsToCommunityPool(ctx sdk.Context, proposalID uint64, totalDeposits []sdk.Coin) error {
+	store := ctx.KVStore(keeper.storeKey)
+
+	proposalCancelBurnRate := keeper.GetParams(ctx).ProposalCancelBurnRate
+	// burn the deposits * proposal_cancel_burn_rate amount from proposal deposits (gov module)
+	burnRate := sdk.MustNewDecFromStr(proposalCancelBurnRate)
+
+	var burnDepositAmount sdk.Coins
+	for _, deposit := range totalDeposits {
+		burnAmount := sdk.NewCoin(
+			deposit.Denom,
+			sdk.NewDecFromInt(deposit.Amount).Mul(burnRate).RoundInt(),
+		)
+		burnDepositAmount = append(burnDepositAmount, burnAmount)
+	}
+
+	// burn the deposits
+	if !burnRate.IsZero() {
+		err := keeper.bankKeeper.BurnCoins(ctx, types.ModuleName, burnDepositAmount)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// send (deposits - burnAmount) to community pool from proposal deposits (gov module)
+	communityPoolAmount := sdk.NewCoins(totalDeposits...).Sub(burnDepositAmount...)
+	err := keeper.dk.FundCommunityPool(ctx, communityPoolAmount, keeper.authKeeper.GetModuleAddress(types.ModuleName))
```

Comment:
nit: we should put `keeper.authKeeper.GetModuleAddress(types.ModuleName)` into a helper method on the `Keeper`. You can call it `ModuleAccountAddress()`